tools/osbuilder/rootfs-builder/rootfs.sh

```diff
@@ -762,6 +770,44 @@ EOF
 	info "Create /etc/resolv.conf file in rootfs if not exist"
 	touch "$dns_file"
 
+    if [ -n "${AA_KBC}" ]; then
+		if [ "${AA_KBC}" == "offline_sev_kbc" ]; then
+			info "Adding agent config for ${AA_KBC}"
+			AA_KBC_PARAMS="offline_sev_kbc::null" envsubst < "${script_dir}/agent-config.toml.in" | tee "${ROOTFS_DIR}/etc/agent-config.toml"
+		fi
+		if [ "${AA_KBC}" == "online_sev_kbc" ]; then
+			info "Adding agent config for ${AA_KBC}"
+			#KBC URI will be specified in the config file via kernel params
+			AA_KBC_PARAMS="online_sev_kbc::123.123.123.123:44444" envsubst < "${script_dir}/agent-config.toml.in" | tee "${ROOTFS_DIR}/etc/agent-config.toml"
+		fi
+		attestation_agent_url="$(get_package_version_from_kata_yaml externals.attestation-agent.url)"
+		attestation_agent_version="$(get_package_version_from_kata_yaml externals.attestation-agent.version)"
+		info "Install attestation-agent with KBC ${AA_KBC}"
+		#git clone "${attestation_agent_url}" --branch "${attestation_agent_tag}" --single-branch
+		git clone --depth=1 "${attestation_agent_url}" guest-components
+
+		pushd guest-components/attestation-agent
+		git fetch --depth=1 origin "${attestation_agent_version}"
+		git checkout FETCH_HEAD
+		( [ "${AA_KBC}" == "eaa_kbc" ] || [ "${AA_KBC}" == "cc_kbc_tdx" ] ) && [ "${ARCH}" == "x86_64" ] && LIBC="gnu"
+		make KBC=${AA_KBC} ttrpc=true
```

Comment:
So, this is a long standing issue that I'd love to get it fixed before we push this into main.

@stevenhorsman has brought this up in the past, as ideally we want to have all the modules enable by default and just load whatever is needed on the runtime side, and I think attesttation-agent should be the one responsible for providing this.

Reply:
thank you @fidencio , I will try to fix it.
And are there more discussion history about it?

Reply:
Hi @LindaYu17 - there are some discussions in https://github.com/kata-containers/kata-containers/issues/8111#issuecomment-1751585623

Reply:
I have updated according to #8111 comments, please help to review the latest two patches of this PR